temporalcli/commands.taskqueue_test.go

```diff
@@ -10,6 +11,107 @@ import (
 	"go.temporal.io/api/workflowservice/v1"
 )
 
+type statsRowType struct {
+	BuildID                 string        `json:"buildId"`
+	TaskQueueType           string        `json:"taskQueueType"`
+	ApproximateBacklogCount int64         `json:"approximateBacklogCount"`
+	ApproximateBacklogAge   time.Duration `json:"approximateBacklogAge"`
+	BacklogIncreaseRate     float32       `json:"backlogIncreaseRate"`
+	TasksAddRate            float32       `json:"tasksAddRate"`
+	TasksDispatchRate       float32       `json:"tasksDispatchRate"`
+}
+
+type taskQueueStatsType struct {
+	Stats []statsRowType `json:"stats"`
+}
+
+func (s *SharedServerSuite) TestTaskQueue_Describe_Task_Queue_Stats_Empty() {
+	// text
+	res := s.Execute(
+		"task-queue", "describe",
+		"--address", s.Address(),
+		"--task-queue", s.Worker().Options.TaskQueue,
+	)
+	s.NoError(res.Err)
+	s.ContainsOnSameLine(res.Stdout.String(), "UNVERSIONED", "workflow", "0", "0s", "0", "0", "0")
+	s.ContainsOnSameLine(res.Stdout.String(), "UNVERSIONED", "activity", "0", "0s", "0", "0", "0")
+
+	// json
+	res = s.Execute(
+		"task-queue", "describe",
+		"--address", s.Address(),
+		"--task-queue", s.Worker().Options.TaskQueue,
+		"-o", "json",
+	)
+	s.NoError(res.Err)
+
+	var jsonOut taskQueueStatsType
+	s.NoError(json.Unmarshal(res.Stdout.Bytes(), &jsonOut))
+	s.Equal(statsRowType{
+		BuildID: "UNVERSIONED",
+		// ordering of workflow/activity pollers is random
+		TaskQueueType:           jsonOut.Stats[0].TaskQueueType,
+		ApproximateBacklogCount: 0,
+		ApproximateBacklogAge:   0,
+		BacklogIncreaseRate:     0,
+		TasksAddRate:            0,
+		TasksDispatchRate:       0,
+	}, jsonOut.Stats[0])
+}
+
+func (s *SharedServerSuite) TestTaskQueue_Describe_Task_Queue_Stats_NonEmpty() {
+	s.Worker().OnDevWorkflow(func(ctx workflow.Context, input any) (any, error) {
+		return map[string]string{"foo": "bar"}, nil
+	})
+
+	// starting a new workflow execution
+	res := s.Execute(
+		"workflow", "start",
+		"--address", s.Address(),
+		"--task-queue", s.Worker().Options.TaskQueue,
+		"--type", "DevWorkflow",
+		"--workflow-id", "my-id1",
+	)
+	s.NoError(res.Err)
+
+	result := s.Execute(
+		"task-queue", "describe",
+		"--address", s.Address(),
+		"--task-queue", s.Worker().Options.TaskQueue,
+	)
+	s.NoError(result.Err)
+	out := result.Stdout.String()
+	s.NotContainsOnSameLine(out, "UNVERSIONED", "workflow", "0", "0s", "0", "0", "0")
```

Comment:
`NotContains` would return true even if nothing is printed. I think we should check the things we want to be shown are shown. (if backlog age is an issue, we should skip that somehow or put a wildcard there).

Also, we should put these checks in EventuallyWithT because the stats may not be updated right away and that'd make the test flaky.

Lastly, I'd check for stats of both wf and activity types, even if activity is expected to have zero stats. This would catch the case if by accident (now or in the future) we print workflow stats for activity as well.